fix(db): --all-features build broken after compile_error removal in #2736

## Description

PR #2736 removed the `compile_error!` macro in `zeph-db/src/lib.rs` that prevented building with both `sqlite` and `postgres` features enabled simultaneously. However, the cfg gates in `pool.rs` were not updated to handle the dual-feature case, resulting in a type mismatch compilation error when building with `--all-features`.

## Reproduction

```bash
cargo build --workspace --all-features
# or
cargo nextest run --workspace --all-features --lib --bins
```

## Error

```
error[E0308]: mismatched types
   --> crates/zeph-db/src/pool.rs:114:12
    |
114 |         Ok(pool)
    |            ^^^^ expected `Pool<Postgres>`, found `Pool<Sqlite>`

error[E0061]: this function takes 2 arguments but 3 were given
  --> crates/zeph-db/src/pool.rs:39:18
```

## Root Cause

In `pool.rs`:
- `connect_sqlite` is gated with `#[cfg(feature = "sqlite")]` instead of `#[cfg(all(feature = "sqlite", not(feature = "postgres")))]`
- The `connect()` method body has both `#[cfg(feature = "sqlite")]` and `#[cfg(feature = "postgres")]` blocks, but when both features are active, both compile and the sqlite block tries to return `Pool<Sqlite>` while `DbPool` resolves to `Pool<Postgres>`

## Expected Behavior

When both `sqlite` and `postgres` features are enabled, postgres should take priority (as documented in `lib.rs`). All cfg gates in `pool.rs` should use `#[cfg(all(feature = "sqlite", not(feature = "postgres")))]` for sqlite-only paths, matching the pattern already used in `migrate.rs` and `transaction.rs`.

## Impact

- `--features full` builds work correctly (only sqlite is active)
- `--all-features` builds fail to compile
- CI matrix with `--all-features` is broken

--- a/crates/zeph-db/src/pool.rs
+++ b/crates/zeph-db/src/pool.rs
@@ -34,7 +34,7 @@ impl DbConfig {
     ///
     /// Returns [`DbError`] if connection or migration fails.
     pub async fn connect(&self) -> Result<DbPool, DbError> {
-        #[cfg(feature = "sqlite")]
+        #[cfg(all(feature = "sqlite", not(feature = "postgres")))]
         {
             Self::connect_sqlite(&self.url, self.max_connections, self.pool_size).await
         }
@@ -44,7 +44,7 @@ impl DbConfig {
         }
     }
 
-    #[cfg(feature = "sqlite")]
+    #[cfg(all(feature = "sqlite", not(feature = "postgres")))]
     async fn connect_sqlite(
         path: &str,
         max_connections: u32,

